src/sentry/api/endpoints/group_similar_issues.py

```diff
@@ -13,10 +13,13 @@
 
 class GroupSimilarIssuesEndpoint(GroupEndpoint):
     def get(self, request, group):
-        # TODO(tkaemming): This should have a limit somewhere.
+        limit = request.GET.get('limit', None)
+        if limit is not None:
+            limit = int(limit) + 1  # the target group will always be included
```

Comment:
I'm probably going to make it so that `compare` no longer considers the search key a valid candidate.

Reply:
I implemented a possible solution here, but I'm not sure how much I like it: 63c37136b16e97e21ccd02a3066b38206eba790b